tests/sweep_framework/sweeps/matmul/full/matmul_default_block_sharded.py

```diff
@@ -138,14 +141,16 @@ def run(
         memory_config=input_b_memory_config,
     )
 
+    start_time = start_measuring_time()
     output_tensor = ttnn.matmul(
         input_tensor_a,
         input_tensor_b,
         memory_config=output_memory_config,
         dtype=output_dtype,
         compute_kernel_config=compute_kernel_config,
     )
+    e2e_perf = stop_measuring_time(start_time)
```

Comment:
@sjameelTT Highlighted that we may want to place the end e2e markers after ttnn.to_torch because some op calls can be asynchronous and won't actually finish until the tensor is pulled from device. Just something to keep in mind.

Reply:
I'll do that.

Reply:
We probably want something that is better than using to_torch going forward since that adds overhead. I know @arakhmati is working on refactoring async so perhaps we can add some api to block until all tensors have been determined.

Reply:
Once that is ready, we can update the tests. For now, this will be sufficient.